internal/pkg/plugininstaller/reposcaffolding/option.go

```diff
@@ -0,0 +1,99 @@
+package reposcaffolding
+
+import (
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/devstream-io/devstream/internal/pkg/plugininstaller"
+	"github.com/devstream-io/devstream/pkg/util/log"
+)
+
+const (
+	transitBranch      = "init-with-devstream"
+	appNamePlaceHolder = "_app_name_"
+)
+
+type Options struct {
+	RepoType        string  `validate:"oneof=gitlab github" mapstructure:"repo_type"`
+	SourceRepo      SrcRepo `validate:"required" mapstructure:"source_repo"`
+	DestinationRepo DstRepo `validate:"required" mapstructure:"destination_repo"`
+	Vars            map[string]interface{}
+}
+
+func NewOptions(options plugininstaller.RawOptions) (*Options, error) {
+	var opts Options
+	if err := mapstructure.Decode(options, &opts); err != nil {
+		return nil, err
+	}
+	return &opts, nil
+}
+
+func (opts *Options) Encode() (map[string]interface{}, error) {
+	var options map[string]interface{}
+	if err := mapstructure.Decode(opts, &options); err != nil {
+		return nil, err
+	}
+	return options, nil
+}
+
+func (opts *Options) CreateAndRenderLocalRepo(workpath string) error {
+	// 1. download template scaffolding repo
+	err := opts.SourceRepo.DownloadRepo(workpath)
+	if err != nil {
+		return err
+	}
+	// 2. walk iter repo files to render template
+	if err := walkLocalRepoPath(workpath, opts); err != nil {
+		log.Debugf("create local repo failed walk: %s.", err)
+		return err
+	}
+	return nil
+}
+
+// PushToRemoteGitLab push local repo to remote gitlab repo
+func (opts *Options) PushToRemoteGitlab(repoPath string) error {
+	// TODO: add gitlab push func
```

Comment:
Maybe it's a good habit to leave your GitHub username here. e.g. form me I'll use: TODO(daniel-hutao): xxx